plugins/builtin/target/aws-asg/plugin/aws_test.go

```diff
@@ -2,15 +2,56 @@ package plugin
 
 import (
 	"errors"
+	"os"
 	"testing"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	hclog "github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/nomad-autoscaler/sdk/helper/scaleutils"
 	"github.com/hashicorp/nomad/api"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestTargetPlugin_SetConfig_NonAWS(t *testing.T) {
+	if os.Getenv("NOMAD_AUTOSCALER_AWS") != "" {
+		t.Skip("skipping aws-asg non-aws environment test, NOMAD_AUTOSCALER_AWS set")
+	}
```

Comment:
this test will only pass where the EC2 metadata is not available; therefore I have added this placeholder as a reminder when adding additional tests that we could use an env var to control which get run depending on the situation.